packages/liveblocks-client/src/room.ts

```diff
@@ -320,21 +320,26 @@ export function makeStateMachine<
         const socket = createWebSocket(rawToken);
         authenticationSuccess(parsedToken, socket);
       } else {
-        return auth(context.roomId)
-          .then(({ token }) => {
-            if (state.connection.state !== "authenticating") {
-              return;
-            }
-            const parsedToken = parseRoomAuthToken(token);
-            const socket = createWebSocket(token);
-            authenticationSuccess(parsedToken, socket);
-            state.token = token;
-          })
-          .catch((er: unknown) =>
-            authenticationFailure(
-              er instanceof Error ? er : new Error(String(er))
+        // token is expired or invalid
+        // therefore we authenticate
+        return (
+          auth(context.roomId)
+            // TODO: provide a meaningful error message when token is undefined here
```

Comment:
It might be easier to throw an error from here https://github.com/liveblocks/liveblocks/blob/eb286928cbdf86472169d30c64162e3ee1891b53/packages/liveblocks-client/src/room.ts#L1752
by wrapping `authentication.callback`. That way the token will never be `undefined` at line 329.

Reply:
done!